File: Binary Search/Medium/LET_no162_Find Peak Element.py
class Solution:
    def findPeakElement(self, nums: List[int]) -> int:
        n = len(nums)
        if n == 1:
            return 0
        
        left, right = 0, len(nums) - 1
        while left + 1 < right:
            mid = (left + right) // 2
            if nums[mid] > nums[mid + 1]:
                right = mid
            else:
                left = mid
        
        if nums[left] > nums[left + 1] and (left == 0 or nums[left - 1] < nums[left]):
            return left
        return right

File: Binary Search/Medium/LET_no162_Find Peak Element.cpp
class Solution {
public:
    int findPeakElement(vector<int>& nums) {
        int n = nums.size();
        if (n == 1)
            return 0;

        int left = 0, right = n - 1;
        while (left + 1 < right) {
            int mid = left + (right - left) / 2;
            if (nums[mid] > nums[mid + 1])
                right = mid;
            else
                left = mid;
        }

        return nums[left] > nums[right] ? left : right;
    }
};